### ISSUE TITLE
Get All Projects > query projects($input: ProjectListInput!) > Getting "Could not find root directory associated to this project" error while rootDirectory is provided in query output.

### ISSUE THREAD

**Issue by iPiyushPatel** [SeedCompany/cord-api-v3#732]
**Describe the bug**
Get All Projects > query projects($input: ProjectListInput!) > Getting "Could not find root directory associated to this project" error while rootDirectory is provided in query output.

**Steps to Reproduce**
On Graphql playground or Postman - create a API request as below snap & attached notepad file and execute request.

**Expected Result**
Project results are returned.

**Actual Result**
Getting "Could not find root directory associated to this project" error and Project results are not returned.
See video - https://drive.google.com/file/d/1_Jz8aYdDOs-LAukjWm7QZdv4vsrdzruI/view?usp=drivesdk

GraphQL Query ( projects ):
![image](https://user-images.githubusercontent.com/5212585/87380418-11b1c380-c5b0-11ea-8daf-d657c355f678.png)

**Query and Input File**
[projects_rootDirectoryERROR.txt](https://github.com/SeedCompany/cord-api-v3/files/4916384/projects_rootDirectoryERROR.txt)

CordAPIV3# 395d59a

**Comment by zkhin** [comment]
@CarsonF This issue occurs because some projects are created without creating its `rootDirectory`
https://github.com/SeedCompany/cord-api-v3/blob/master/src/components/project/project.service.ts#L592~L595

So I suggest to refactor `getRootDirectory` like below.
```
if (!rootRef?.id) {
      const project = await this.readOne(projectId, session);
      await this.eventBus.publish(new ProjectCreatedEvent(project, session));

      rootRef = await this.db
        .query()
        .match(matchSession(session, { withAclRead: 'canReadProjects' }))
        .match([
          [
            node('project', 'Project', { active: true, id: projectId }),
            relation('out', 'rootDirectory', { active: true }),
            node('directory', 'BaseNode:Directory'),
          ],
        ])
        .return({
          directory: [{ id: 'id' }],
        })
        .first();
}
```
If the `rootRef.id` does not exist, it will create a project's `rootDirectory`.
Let me get your feedback.

**Comment by CarsonF** [comment]
Querying an object should not cause a mutation (a change in the database).

We need more information about how that project was created, with what user, etc. to determine the problem.
We don't know if the directory is being created or not on the mutation or whether it's a read/permission problem on the query.

**Comment by zkhin** [comment]
I see... so can you let me know what is the best solution to cover this issue?

**Comment by CarsonF** [comment]
I don't know what the root problem is. How is it reproduced? What's the state of the db after the project is created? Is there a `rootDirectory` relationship?

**Comment by zkhin** [comment]
It happens in the `list` query. You can reproduce with @iPiyushPatel 's query above.
It only happens when the project does not have `rootDirectory`

**Comment by CarsonF** [comment]
OH. The local db just needs to be wiped then.
We've fixed this with #708.
